app/controllers/incomeSources/add/IncomeSourceAddedController.scala

```diff
@@ -0,0 +1,164 @@
+            for {
+              businessName <- addedBusiness.tradingName
+              startDate <- addedBusiness.tradingStartDate
+            } yield (startDate, Some(businessName))
+          }
+      case UkProperty =>
+        for {
+          newlyAddedProperty <- user.incomeSources.properties.find(incomeSource =>
+            incomeSource.incomeSourceId.equals(incomeSourceId) && incomeSource.isUkProperty
+          )
+          startDate <- newlyAddedProperty.tradingStartDate
+        } yield (startDate, None)
+      case ForeignProperty =>
+        for {
+          newlyAddedProperty <- user.incomeSources.properties.find(incomeSource =>
+            incomeSource.incomeSourceId.equals(incomeSourceId) && incomeSource.isForeignProperty
+          )
+          startDate <- newlyAddedProperty.tradingStartDate
+        } yield (startDate, None)
+    }
+  }
+
+  private def handleRequest(isAgent: Boolean, incomeSourceId: String, incomeSourceType: IncomeSourceType)(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+    withIncomeSourcesFS {
+      val backUrl = getBackUrl(incomeSourceId, isAgent, incomeSourceType)
+      getIncomeSourceData(incomeSourceType, incomeSourceId) match {
+        case Some(incomeSourceData) =>
+          val showPreviousTaxYears: Boolean = incomeSourceData._1.isBefore(dateService.getCurrentTaxYearStart())
+          incomeSourceType match {
+            case SelfEmployment =>
+              incomeSourceData._2 match {
+                case Some(businessName) => nextUpdatesService.getObligationsViewModel(incomeSourceId, showPreviousTaxYears) map { viewModel =>
+                  Ok(obligationsView(businessName = Some(businessName), sources = viewModel, backUrl = backUrl, isAgent = isAgent, incomeSourceType = SelfEmployment))
+                }
+                case None => nextUpdatesService.getObligationsViewModel(incomeSourceId, showPreviousTaxYears) map { viewModel =>
+                  Ok(obligationsView(sources = viewModel, backUrl = backUrl, isAgent = isAgent, incomeSourceType = SelfEmployment))
+                }
+              }
+            case UkProperty => nextUpdatesService.getObligationsViewModel(incomeSourceId, showPreviousTaxYears) map { viewModel =>
+              Ok(obligationsView(viewModel, backUrl, isAgent = isAgent, incomeSourceType = UkProperty))
+            }
+            case ForeignProperty => nextUpdatesService.getObligationsViewModel(incomeSourceId, showPreviousTaxYears) map { viewModel =>
+              Ok(obligationsView(viewModel, backUrl, isAgent = isAgent, incomeSourceType = ForeignProperty))
+            }
+          }
+        case None => Logger("application").error(
+          s"[IncomeSourceAddedController][handleRequest] - unable to find incomeSource by id: $incomeSourceId")
+          if (isAgent) Future(itvcErrorHandlerAgent.showInternalServerError())
+          else Future(itvcErrorHandler.showInternalServerError())
+      }
```

Comment:
if you can add recover for the resulting Future of this method, as we should handle critical error as well by logging them etc